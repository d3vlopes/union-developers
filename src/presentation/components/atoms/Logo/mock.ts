import { LogoProps } from '.'

export const logoMock: LogoProps = {
  imgUrl:
    'https://res.cloudinary.com/dbnq26wqe/image/upload/v1687995701/union/logo_desktop_egwas8.png',
  alt: 'Um octógono com três pontos, representando a união dos desenvolvedores',
}
