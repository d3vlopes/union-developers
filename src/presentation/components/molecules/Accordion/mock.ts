import { AccordionProps } from '.'

export const accordionMock: AccordionProps = {
  title: 'O que é o Union?',
  content: `
    <p>O Union é um ambiente prático e imersivo próximo do encontrado nas empresas, desafiando seus membros a resolver problemas reais, com cronogramas e prazos de entrega.</p> 
    <p>No Union, você desenvolve as skills mais procuradas pelo mercado, trabalha com ferramentas e metodologias utilizada pelas empresas, além de ganhar experiência trabalhando com outras pessoas de varias partes do Brasil.</p> 
    <p>Você vai vivenciar na prática como é o dia a dia de um desenvolver em uma empresa e ficar mais preparado para conquistar sua vaga.</p>
  `,
}
