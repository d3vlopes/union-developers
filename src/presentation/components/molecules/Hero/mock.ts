import { HeroProps } from './types'

type MockType = 'default' | 'long-title' | 'long-description'

const defaultProps: HeroProps = {
  title:
    '<h1>Quer se tornar um desenvolvedor <strong>diferenciado?</strong></h1>',
  description:
    'Aqui você vivencia na prática como é o dia a dia de um desenvolvedor em uma empresa e se prepara para enfrentar os desafios do mercado de forma prática e imersiva!',
  buttonText: 'Quero participar',
  buttonTarget: '#join',
  backgroundMobile:
    'https://media.graphassets.com/j1rOxE97RsCbeyFrMU4z',
  backgroundDesktop:
    'https://media.graphassets.com/66rHD5NBR0K9lcIF9t62',
  logoUrl: 'https://media.graphassets.com/zRjyxnD5RyGdG8xVpll8',
  logoAlt: 'Description logo',
}

export const heroMockFactory: Record<MockType, HeroProps> = {
  default: {
    ...defaultProps,
  },
  'long-title': {
    ...defaultProps,
    title:
      '<h1>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!</h1>',
  },
  'long-description': {
    ...defaultProps,
    description: `
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!
    `,
  },
}
