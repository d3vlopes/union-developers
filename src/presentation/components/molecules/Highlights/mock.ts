import { HighlightModel } from '@/domain/models'

export const highlightsMock: HighlightModel[] = [
  {
    id: '1',
    icon: {
      url: 'https://media.graphassets.com/qssZQEUSEaAQoM6oznCw',
      alt: 'rocket',
    },
    title: 'Crescimento profissional',
    description:
      'Eleve o seu conhecimento, aperfeiçoe suas habilidades e alavanque o seu potencial com uma experiência intensa que te ajudará a ser um profissional ainda melhor.',
  },
  {
    id: '2',
    icon: {
      url: 'https://media.graphassets.com/BN50QqvcR67hdjNyuqkA',
      alt: 'pencil and scale',
    },
    title: 'Aumente sua caixa de ferramentas',
    description:
      'Descubra novas metodologias e encare desafios que agregam em seu conhecimento profissional, essenciais para avançar na sua trajetória.',
  },
  {
    id: '3',
    icon: {
      url: 'https://media.graphassets.com/WR1yF2qGRmm8jWLCyw0D',
      alt: 'like',
    },
    title: 'Problemas Reais',
    description:
      'Trabalhe em projetos reais com equipes reais para sair da sua zona de conforto e acelerar seu aprendizado.',
  },
  {
    id: '4',
    icon: {
      url: 'https://media.graphassets.com/2S5Pdk5oShOJUlIz8p7X',
      alt: 'planet',
    },
    title: 'Ambiente Imersivo',
    description:
      'Espaço completo que te prepara, conecta com o mercado e permite absorver conteúdos fundamentais para seguir a sua jornada profissional.',
  },
  {
    id: '5',
    icon: {
      url: 'https://media.graphassets.com/KHCFnl40RyKQqOx6Bjuz',
      alt: 'bulb on',
    },
    title: 'Experiência',
    description:
      'Com os processos fundamentais e domínio das ferramentas te dá a experiência necessária para tomar as melhores decisões no mundo real.',
  },
  {
    id: '6',
    icon: {
      url: 'https://media.graphassets.com/XH5lk8IJQ1Cz1M2AE1Lu',
      alt: 'game controller',
    },
    title: 'Vivência na Prática',
    description:
      'Vivencie o dia a dia do mercado com um time de diferentes mentalidades para solucionar problemas em diversas áreas.',
  },
  {
    id: '7',
    icon: {
      url: 'https://media.graphassets.com/eBMuKBvTLulsEbZySGwt',
      alt: 'star',
    },
    title: 'Desenvolvimento de Skills',
    description:
      'Aprenda a comunicar suas ideias em um ambiente profissional e melhorar suas habilidades de gerenciamento de tempo para se destacar em sua carreira.',
  },
  {
    id: '8',
    icon: {
      url: 'https://media.graphassets.com/sanArFzbQUipTnjRVl78',
      alt: 'person',
    },
    title: 'Mentoria',
    description:
      'No Union você irá ter do lado um profissional experiente que vai te guiar e mostrar os atalhos para que você consiga evoluir mais rápido em menos tempo.',
  },
]
