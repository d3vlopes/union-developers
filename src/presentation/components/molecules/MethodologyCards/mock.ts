import { MethodologyModel } from '@/domain/models'

export const methodologyCardsMock: MethodologyModel[] = [
  {
    id: '1',
    title: 'Ativa',
    description:
      'É uma metodologia de aprendizado que coloca o aluno como protagonista, onde ele é o responsável por ir atrás dos conhecimentos necessários, ficando o professor como um mentor, que vai mostrar o caminho e indicar os melhores conteúdos, que vai guiar o estudo e os métodos, mas também deixar espaço para o aluno explorar.',
  },
  {
    id: '2',
    title: 'Andragogia',
    description:
      'É uma metodologia de ensino para adultos, que tem mais liberdade e colaboração, permitindo que o aluno decida quando, como e o que estudar, tornando o processo de aprendizagem mais sólido e eficaz. É uma metodologia de ensino mais prática, que estimula o aluno a resolver problemas reais, que estão relacionado ao seu interesse.',
  },
  {
    id: '3',
    title: 'Ágeis',
    description:
      'A metodologia ágil é uma metodologia que busca entregar valor a todo momento, estimula uma maior colaboração entre o time, permitindo que tenha uma visão mais transparente, tendo liberdade para fazer escolhas e tomar decisões, também estimula a melhora continua entre os membros, processos e da qualidade dos entregáveis.',
  },
]
