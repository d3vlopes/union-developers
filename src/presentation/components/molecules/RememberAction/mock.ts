import { RememberActionProps } from '.'

export const rememberActionMock: RememberActionProps = {
  title:
    'Vivencia na prática, ambiente imersivo, desafios e muito mais!',
  subtitle: 'Inscreva-se já! Vagas limitadas',
  buttonText: 'Inscrever-se agora',
  buttonTarget: '#subscribe',
  logoUrl: 'https://media.graphassets.com/WayysMB8RQyrgcahBMQs',
  logoAlt: 'Logo description',
}
