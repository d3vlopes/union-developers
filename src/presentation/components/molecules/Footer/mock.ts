import { generateId } from '@/utils'

import { logoMock } from '@/presentation/components/atoms/Logo/mock'

import { FooterProps } from '.'

export const footerMock: FooterProps = {
  logo: {
    imgUrl: logoMock.imgUrl,
    alt: logoMock.alt,
  },
  socialLinks: [
    {
      id: generateId(),
      icon: {
        imgUrl: 'https://media.graphassets.com/4AjPcOSSuK4kIKUv8Mdg',
        alt: 'Um ícone de envelope representando o email do Union',
      },
      label: 'Email',
      target: 'contato.uniongroup@gmail.com',
    },
    {
      id: generateId(),
      icon: {
        imgUrl: 'https://media.graphassets.com/y4KMQdygSH6uZx46pxjU',
        alt: 'Ícone com as letras in representando a logo do Linkedin',
      },
      label: 'Linkedin',
      target: 'www.linkedin.com/in/company/uniondevelopers',
    },
    {
      id: generateId(),
      icon: {
        imgUrl: 'https://media.graphassets.com/4lDgbi8FQpq8QYSiranx',
        alt: 'Um gatinho representando o logo do Github',
      },
      label: 'Github',
      target: 'www.github.com/union-group',
    },
  ],
  companyName: 'Union Developers',
  companySlogan:
    'O Union proporciona experiências para desenvolvedores em inicio de carreira, desafiando seus membros a resolver problemas reais.',
  navigationMenuLinks: [
    {
      id: generateId(),
      title: 'Saiba mais',
      links: [
        {
          label: 'Sobre',
          target: '#about',
        },
        {
          label: 'Depoimentos',
          target: '#testimonials',
        },
      ],
    },
    {
      id: generateId(),
      title: 'Dúvidas',
      links: [
        {
          label: 'FAQ',
          target: '#questions',
        },
        {
          label: 'Como funciona',
          target: '#steps',
        },
      ],
    },
  ],
}
