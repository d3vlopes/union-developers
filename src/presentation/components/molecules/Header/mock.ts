import { logoMock } from '@/presentation/components/atoms/Logo/mock'

import { HeaderProps } from './types'

export const headerMock: HeaderProps = {
  logoUrl: logoMock.imgUrl,
  logoAlt: logoMock.alt,
  navigationLinks: [
    {
      label: 'Sobre',
      target: '#about',
    },
    {
      label: 'Benefícios',
      target: '#highlights',
    },
    {
      label: 'Etapas',
      target: '#steps',
    },
    {
      label: 'Dúvidas',
      target: '#questions',
    },
  ],
  buttonText: 'Junte-se a nós',
  buttonTarget: '#join',
}
