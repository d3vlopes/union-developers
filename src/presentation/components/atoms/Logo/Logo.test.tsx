import { render, screen } from '@/__tests__/helpers'

import { logoMock } from './mock'

import { Logo } from '.'

const { getByRole } = screen

describe('<Logo />', () => {
  it('should correct alt', () => {
    render(<Logo {...logoMock} />)

    const logo = getByRole('img')

    expect(logo).toHaveAttribute('alt', logoMock.alt)
  })

  it('should correct src', () => {
    render(<Logo {...logoMock} />)

    const logo = getByRole('img')

    expect(logo).toHaveAttribute('src', logoMock.imgUrl)
  })

  it('should render <a> element around <img> with correct href for target', () => {
    render(<Logo {...logoMock} />)

    const link = getByRole('link')

    expect(link).toHaveAttribute('href', '#')
  })

  it('should match snapshot', () => {
    const { container } = render(<Logo {...logoMock} />)

    expect(container).toMatchSnapshot()
  })
})
