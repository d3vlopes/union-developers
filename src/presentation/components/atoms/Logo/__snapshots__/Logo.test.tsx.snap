// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Logo /> > should match snapshot 1`] = `
@media (max-width:1100px) {
  .c0 img {
    width: 22rem;
  }
}

<div>
  <div
    class="c0"
  >
    <a
      href="#"
    >
      <img
        alt="Um octógono com três pontos, representando a união dos desenvolvedores"
        src="https://res.cloudinary.com/dbnq26wqe/image/upload/v1688490290/union/logo_desktop_uppabl.svg"
      />
    </a>
  </div>
</div>
`;
